lib/src/main/java/org/team1540/rooster/drive/pipeline/TankDriveData.java

```diff
@@ -0,0 +1,68 @@
+package org.team1540.rooster.drive.pipeline;
+
+import java.util.Objects;
+import java.util.OptionalDouble;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Encapsulates drive commands for a tank drive.
+ *
+ * {@code TankDriveData} instances are usually passed around in drive pipelines. Each {@code
+ * TankDriveData} instance contains {@link DriveData} instances for the left and right side, as well
+ * as commands for {@linkplain #heading} and {@linkplain #turningRate turning rate}.
+ *
+ * @see DriveData
+ */
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
```

Comment:
Is that the right way, or should we just let nulls run around? I feel like the latter is actually correct, despite how much I dislike it.

Reply:
Well primitives aren’t nullable and if we’re gonna box things may as well have a nice thing for it.